src/commonMain/kotlin/au/id/micolous/metrodroid/transit/intercard/IntercardTransitData.kt

```diff
@@ -0,0 +1,84 @@
+/*
+ * OpalTransitData.java
+ *
+ * Copyright 2015-2018 Michael Farrell <micolous+git@gmail.com>
+ *
+ * This program is free software: you can redistribute it and/or modify
+ * it under the terms of the GNU General Public License as published by
+ * the Free Software Foundation, either version 3 of the License, or
+ * (at your option) any later version.
+ *
+ * This program is distributed in the hope that it will be useful,
+ * but WITHOUT ANY WARRANTY; without even the implied warranty of
+ * MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ * GNU General Public License for more details.
+ *
+ * You should have received a copy of the GNU General Public License
+ * along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+package au.id.micolous.metrodroid.transit.intercard
+
+import au.id.micolous.metrodroid.transit.*
+
+import au.id.micolous.metrodroid.card.CardType
+import au.id.micolous.metrodroid.card.desfire.DesfireCard
+import au.id.micolous.metrodroid.card.desfire.DesfireCardTransitFactory
+import au.id.micolous.metrodroid.card.desfire.settings.ValueDesfireFileSettings
+import au.id.micolous.metrodroid.multi.Parcelize
+import au.id.micolous.metrodroid.multi.R
+import au.id.micolous.metrodroid.ui.ListItem
+
+@Parcelize
+class IntercardTransitData (
+        private val mSerialNumber: Long,
+        private val mBalance: Int?, // 10th of cents
+        private val mLastTransaction: Int?
+): TransitData() {
+
+    override val cardName: String
+        get() = NAME
+
+    public override val balance
+        get() = mBalance?. let { parseCurrency(it) }
+
+    override val serialNumber: String?
+        get() = mSerialNumber.toString()
+
+    override val info: List<ListItem>? get() = listOfNotNull(
+        mLastTransaction?.let { ListItem(R.string.last_transaction, parseCurrency(mLastTransaction).formatCurrencyString(true))}
+    )
+
+    companion object {
+        const val NAME = "Intercard"
+        const val APP_ID_BALANCE = 0x5f8415
+
+        private val CARD_INFO = CardInfo(
+                name = NAME,
+                locationId = R.string.location_germany_and_switzerland,
+                cardType = CardType.MifareDesfire)
+
+        // FIXME: Apparently this system may be either in euro or in Swiss Francs. Unfortunately
+        // Swiss Franc one still has string "EUR" in file 0, so this suggests a lazy adaptation.
+        // I don't have samples from Germany to compare.
+        private fun parseCurrency(input: Int): TransitCurrency =  TransitCurrency(input / 10, "CHF")
```

Comment:
I did a little digging, it looks like the same as what is supported by [MenzaGuthaben](https://github.com/jakobwenzel/MensaGuthaben).  That even adopted DESFire reading code from FareBot!  There's another system it supports (MagnaCarta) which looks easy to get.

Reading that code, it doesn't look like there is any code differences to detect the university name, [but there are many that use one of these two systems](https://github.com/jakobwenzel/MensaGuthaben/tree/master/docs#working-universities).

I think it's unlikely there's a difference between cards in CHF or EUR.  For display purposes, this could be a preference (`intercard_currency`), defaulted to EUR (as the population of Germany is ~10x higher).

If we wanted to be fancy, the default state could switch to `CHF` if the phone's MCC = 228.  I think that should be possible with locale files without needing to get telephony permission.

Reply:
Oh, you already found MenzaGuthaben -- I hadn't read #439 yet, blah.

Reply:
Added this to the comment. Basically there is another open file, 32 bytes long, 7 bytes filled + checksum. I think ideally we should read currency from file 0 but have a list of overrides from university id. But again we need more samples.